pkg/apis/addon/v1alpha1/addon_types.go

```diff
@@ -292,6 +294,8 @@ type AddonStatus struct {
 // +kubebuilder:printcolumn:name="STATUS",type="string",JSONPath=".status.lifecycle.installed"
 // +kubebuilder:printcolumn:name="REASON",type="string",JSONPath=".status.reason"
 // +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+// +genclient
```

Comment:
Addon-manager and most other Keikoproj controllers were generated by **kubebuilder@v2** and according to the migration [docs](https://book.kubebuilder.io/migration/legacy/migration_guide_v1tov2.html#migrate-the-apis) 

> We don’t need the following markers any more (they’re not used anymore, and are relics from much older versions of KubeBuilder):
// +genclient
// +k8s:openapi-gen=true

Addon-manager should remain aligned to v2 of Kubebuilder until we upgrade to v3, but why are these comment markers?
 
I also have concern about removing `/api` directory as that is required by **kubebuilder** scaffolding

Reply:
Those "comment marker" works for code-generator. And it does not impact kubebuilder. Believing as long as we do not remove kubebuilder comments, kubebuilder v2/v3 should work as expected. 
both code-generator and kubebuilder has options specifying API directory, do not see any concern here.

Reply:
In that case not sure why we'd choose to remove `/api` if you can generate to any directory, the design appears to resemble [sample-controller](https://github.com/kubernetes/sample-controller/tree/master/pkg/apis/samplecontroller/v1alpha1) which pre-dates **kubebuilder**, likely the genesis of kubebuilder@v0 and we're in v2.

I don't see the reason to remove `/api` even if you want to just have `/pkg/apis`.

Reply:
Move api and client to a central place or not is not a blocker at all.  
Not quite get the comparison between kubebuilder and workqueue (sample, argo etc used), I do not think they have any conflicts.

Reply:
Argo workflow-controller pre-dates **kubebuilder** and used **sample-controller** design, this is not the same as  **kubebuilder** which has an opinion about generated clientset... See https://github.com/kubernetes-sigs/kubebuilder/issues/403#issuecomment-518437978
> "Libraries Over Code Generation"

https://github.com/kubernetes-sigs/kubebuilder/blob/master/DESIGN.md#overarching

What we're saying is if we want to generate a clientset let's not change the opionated way of kubebuilder but instead do it separately; there's no requirement to remove `/api`.

Reply:
You can have both `/api` and `/pkg/apis` as separate things, correct?

Reply:
We use clientset/lister/informer accessing addon API and thus react (install workflow, update status) directly. with so, reduce cpu/mem, and process cycle.
We only need maintain one Addon API in the repo, we not allowed duplicate definition

Reply:
I am against moving things to `/pkg` if you change to use top-level `/apis` directory I will overlook the pathing.

For the rest of this PR I expect this code generator to be tied into the build, either via go generate or through the makefile.

See how someone else did similar.
https://www.fatalerrors.org/a/writing-crd-by-mixing-kubeuilder-and-code-generator.html

Reply:
This is an awesome idea, will follow.

Reply:
Fixed https://github.com/keikoproj/addon-manager/pull/111/commits/d05c872afa597828065cea15a2178ad981f65dd0